### ISSUE TITLE
Feature Request: Hide illegal moves from move list

### ISSUE THREAD

**Issue by Warslach** [kwsch/PKHeX#341]
There needs to be an option to only allow moves the pokemon can actually learn to show up instead of every move ever. I know it is like that because some pokemon can learn 'event moves' but I think everyone could do with an option that only allows moves pokemon can legally learn for battle. Even an option that by default shows only the moves the pokemon learns and if you tick a box it shows all moves ever that can be chosen.

Thanks

**Comment by ObscenityIB** [comment]
hmm, yeah, disableable option would be good, but not selected by default

**Comment by Ninjistix** [comment]
honestly I was thinking requesting something like this as "move learn sets" which would make it easier to generate "legit" pkmn, and also if possible a DB of legit "meet locations" depending on pkmn (if its possible I would gladly help fill/contribute information into the DB)

**Comment by ricksee** [comment]
I also recommend something like this.

**Comment by kwsch** [comment]
Probably not happening the way you've requested, as loading 'mons with 'illegal' moves would not appear correctly as the move list is filtered.

Then comes the question on how the program reacts when changing species with a movepool already existing. Does it clear? Beyond species, certain forms have access to moves while others do not. What happens when the form is changed, or the origin game? What happens if the Pokemon was migrated from a previous game, and the level up table has changed between generations?

Currently the program tells you if a choice is invalid after selecting.
A pre-check is flawed as certain encounter circumstances lead to certain moves being available, ie Azurill and Marill having slightly different egg move tables.

In conclusion, it seems the best solution is to not filter the dropdown but pre-indicate if certain moves are possible would be feasible yet limited to moves obtainable in the current generation. Would probably have to expand the Legality core to return getValidMoves and getValidRelearn, and use those arrays for visual indication in the dropdowns (override DrawItem to paint a different color if present).

That way the full movepool is still visible, yet it's easily discernible what moves it has access to in the current format.

**Comment by Ninjistix** [comment]
thanks as always @kwsch
